tests/hierarchy_driver.cc

```diff
@@ -63,33 +64,48 @@ void matrix_free_two_grids(std::shared_ptr<boost::property_tree::ptree> params)
       std::make_shared<TestMFMeshEvaluator<dim, fe_degree, double>>(
           mf_laplace._dof_handler, mf_laplace._constraints,
           mf_laplace._laplace_operator, material_property);
-  mfmg::Hierarchy<DVector> hierarchy(comm, evaluator, params, timer);
-
-  // We want to do 20 V-cycle iterations. The rhs of is zero.
-  // Use D(istributed)Vector because deal has its own Vector class
-  DVector residual(rhs);
-  unsigned int const n_cycles = 20;
-  std::vector<double> res(n_cycles + 1);
 
-  mf_laplace._laplace_operator.vmult(residual, solution);
-  residual.sadd(-1., 1., rhs);
-  auto const residual0_norm = residual.l2_norm();
+  mfmg::Hierarchy<DVector> hierarchy(comm, evaluator, params, timer);
 
-  std::cout << std::scientific;
-  res[0] = 1.0;
-  for (unsigned int i = 0; i < n_cycles; ++i)
+  if (!test_preconditioner)
```

Comment:
I don't find `"is preconditioner"` not very descriptive.  Please comment that it let the user specify whether to solve the Laplace problem and report the number of iterations required or whether to perform a set number of V-cycles and report the convergence rate.

Reply:
"is preconditioner" is preexisting. Its main purpose is to decide whether to zero out the destination vector in `Hierarchy::apply`.